docs/baseplate/core.rst

```diff
@@ -1,19 +1,62 @@
-================================
-Diagnostics: Spans and Observers
-================================
+baseplate.core
+==============
 
-Spans
------
+The heart of the Baseplate framework is its diagnostics system. Here's an
+incomplete example of an application built on the framework::
+
+   def do_something(request):
+       request.some_redis_client.ping()
+
+   def make_app(app_config):
+       ... snip ...
+
+       baseplate = Baseplate()
+       baseplate.configure_metrics(metrics_client)
+       baseplate.add_to_context(
+           "some_redis_client", RedisContextFactory(redis_pool))
+
+       ... snip ...
 
-The heart of Baseplate is its diagnostics system.
+When a request is made which routes to the ``do_something`` handler, a
+:py:class:`~baseplate.core.RootSpan` is automatically created. If the incoming
+request has trace headers, the root span will be identical to the child span in
+the upstream service. When we call ``request.some_redis_client.ping()`` in the
+handler, Baseplate will create a child :py:class:`~baseplate.core.Span` object
+to represent the time taken talking to redis.
```

Comment:
Nice!  Definitely getting a better grasp of it now.

While I think I get the gist of most of this, what does it mean for a Span object to be created?  Sure, it represents some time taken, but what does it do?  Is that Span object then processed in some way later?  Does it automatically perform some action when it's created?  Or does it just keep the information there to be accessed later?

I see below you say the span nonsense means baseplate automatically sends metrics based on them.  Is this through the observers?

It's getting much clearer! :-)  All of this might be explained below, but since you're adding an explanation here, it feels *almost* complete for a high-level overview.